python_modules/libraries/dagster-dask/dagster_dask/executor.py

```diff
@@ -246,7 +246,13 @@ def execute(self, plan_context, execution_plan):
                         for key in step_input.dependency_keys:
                             dependencies.append(execution_futures_dict[key])
 
-                    run_config = dict(plan_context.run_config, execution={"in_process": {}})
+                    # If the mode name is the default mode name for a job, then we can be
+                    # reasonably confident that we are executing a job, in which case it doesn't
+                    # make sense to be able to switch on in_process execution.
+                    if plan_context.pipeline_run.mode == "default":
+                        run_config = plan_context.run_config
+                    else:
+                        run_config = dict(plan_context.run_config, execution={"in_process": {}})
```

Comment:
the `default` is also what you get if you don't specify explicitly your modes on a pipeline so this is definitely dubious

what is the context for this change?

Reply:
It is indeed, but that `default` mode would never get to this code, because it's not using the dask executor, no?

Context is that a user was running into this error:
```Error 1: Received unexpected config entry "in_process" at path root:execution. Expected: "{ config: { cluster: { existing: { address: (String | { env: String }) } kube?: { } local?: { } lsf?: { } moab?: { } oar?: { } pbs?: { } sge?: { } slurm?: { } ssh?: { } yarn?: { } } } }".
    Error 2: Missing required config entry "config" at path root:execution. Sample config for missing entry: {'config': {'cluster': '<selector>'}}```

When providing the following config, which should be valid:
```yaml
execution: 
  config:
    cluster:
      pbs:
        cores: 28
        memory: 120gb
        n_workers: 1
        project: dagster
        queue: mpi_4
        resource_spec: "select=4:ncpus=28:mem=120gb"
        walltime: "00:05:00"
```

Reply:
@alangenfeld to be clear though, I agree that it's dubious. The right approach here is probably to thread through information about whether or not we're using the dask executor.

Reply:
>  because it's not using the dask executor, no?

`name` is optional on `ModeDefinition` so if you don't set it you will have `default`, which can be the case if you are only setting the executors on it. 

Can you not derive "jobliness" from some other piece of information on the execution plan or context?